### PR TITLE
Swagger completeness for Redis Cache

### PR DESCRIPTION
**Pull request by SiddharthChatrolaMs**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/redis/resource-manager/Microsoft.Cache/stable/2017-10-01/redis.json [hunk 1]
```diff
@@ -59,6 +59,90 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Cache/CheckNameAvailability": {
+      "patch": {
```
Comment (ravbhatnagar) [inline specification/redis/resource-manager/Microsoft.Cache/stable/2017-10-01/redis.json]: This should be a POST instead of a PATCH

**Comment by ravbhatnagar** [comment]
@SiddharthChatrolaMs - one comment

**Comment by SiddharthChatrolaMs** [comment]
@ravbhatnagar Fixed comments

File: specification/redis/resource-manager/Microsoft.Cache/stable/2017-10-01/redis.json (hunk 1 above)
Comment (SiddharthChatrolaMs) [reply specification/redis/resource-manager/Microsoft.Cache/stable/2017-10-01/redis.json]: Done

**Comment by SiddharthChatrolaMs** [comment]
Can someone please merge this?

**Comment by lmazuel** [comment]
@SiddharthChatrolaMs please fix the Travis job `model PR_ONLY=true`

**Comment by SiddharthChatrolaMs** [comment]
@lmazuel: "please fix the Travis job model PR_ONLY=true" => what is that i need to fix here? I see the build passed "continuous-integration/travis-ci/pr — The Travis CI build passed"

**Review by lmazuel — APPROVED** [review]

**Pull request merged**